src/domains/chart/sagas.ts

```diff
@@ -3,67 +3,93 @@ import {
   put,
   takeEvery,
   select,
+  spawn,
+  take,
 } from "redux-saga/effects"
+import { channel } from "redux-saga"
 
 import { axiosInstance } from "utils/api"
 import { alwaysEndWithSlash } from "utils/server-detection"
+import { fetchMetricsStream } from "utils/netdata-sdk"
 
 import { selectGlobalPanAndZoom } from "domains/global/selectors"
 import { StateT as GlobalStateT } from "domains/global/reducer"
+
 import {
   fetchDataAction, FetchDataPayload,
   fetchChartAction, FetchChartPayload,
 } from "./actions"
 
+const fetchDataResponseChannel = channel()
+export function* watchFetchDataResponseChannel() {
+  while (true) {
+    const action = (yield take(fetchDataResponseChannel))
+
+    // special case - if requested relative timeRange, and during request the mode has been changed
+    // to absolute global-pan-and-zoom, cancel the store update
+    // todo do xss check of data
+    if (action.type === fetchDataAction.success.toString()) {
+      const { viewRange } = action.payload.fetchDataParams
+      const globalPanAndZoom = (yield select(
+        selectGlobalPanAndZoom,
+      )) as GlobalStateT["globalPanAndZoom"]
+
+      if (globalPanAndZoom
+        && (viewRange[0] <= 0 || viewRange[1] <= 0)
+      ) {
+        // eslint-disable-next-line no-continue
+        continue
+      }
+    }
+
+    yield put(action)
+  }
+}
+
+
 type FetchDataSaga = { payload: FetchDataPayload }
```

Comment:
I think we could use common way to declare actions, like

```type MyAction<T = any> = { action: string, payload: T}```

In this case you will declare only generic in arguments. Here will be

```
function fetchDataSaga({ payload }: Action<FetchDataPayload>) {
```

Reply:
good idea, i've reused "redux-act" `Action` type